keps/sig-storage/596-csi-inline-volumes/README.md

```diff
@@ -213,32 +250,214 @@ Inline volume requests can only participate in mount/unmount volume operations.
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+
+  There are no new dependencies introduced by this feature.
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+  There will be no new API calls. A new API field is added to the CSIDriver
+  API to determine if the driver supports ephemeral inline volumes.
+
+###### Will enabling / using this feature result in introducing new API types?
+
+  The only new API type is the `VolumeLifecycleMode` string that is used
+  to specify supported modes in the CSIDriver spec.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+  No, ephemeral inline volumes can only participate in mount/unmount volume operation
+  calls to the CSI driver.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+  There is a new `VolumeLifecycleModes` field in CSIDriverSpec that may slightly increase
+  the size by adding 2 new strings (`Persistent` and `Ephemeral`).
+
+  There is also `pod.spec.volumes` that can contain `csi` items now. Its size is comparable
+  to the size of Secrets/ConfigMap volumes inline in Pods. The CSI volume definitions will
+  be bigger, as they contain a generic map of `volumeAttributes`, which contains opaque
+  parameters for the CSI driver.
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+  Compared to CSI persistent volumes, there should be no increase in the amount of time
+  taken to mount / unmount, as there will be fewer CSI calls required for inline volumes.
+  Compared to Secrets and ConfigMaps, inline CSI volumes will be slower mount / unmount,
+  since an external CSI driver is responsible for providing the actual volume.
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+  Kubernetes itself should not see any noticeable increase in resource consumption,
+  but CSI driver pods will need to be deployed on all the nodes in order to make
+  use of this feature.
+
+### Troubleshooting
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+  Pods will not start and volumes for them will not get provisioned.
+
+###### What are other known failure modes?
+
+  If the storage system fails to provision inline volumes, there will
+  be an event in the affected pod indicating what went wrong when
+  mounting the CSI volume.
+
+  Failure modes related to individual CSI drivers may require examining
+  the CSI driver logs.
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+  The kubelet and CSI driver logs should be inspected to look for errors mounting / unmounting the volume, or to help explain why operations are taking longer than expected.
```

Comment:
Any example logs to look for? What we log when there is a failure? As an operator, what I should be looking for?